### PR TITLE
Pylint testing for travis

### PR DESCRIPTION
**Pull request by pgdr**
This PR adds linting as a test case.

### CODE REVIEW

**Comment by pgdr** [comment]
Beware that the last commits deletes a function `Fault.extendToFault(self, other_fault, k)` as it was defined twice.

**Comment by pgdr** [comment]
Beware that this PR also temporarily disables clang testing.  If this PR is merged, a task will be created for proper clang support.

**Comment by pgdr** [comment]
Removal of clang also takes build+test on travis from 6m20s to ~3m.

File: python/python/ert/ecl/faults/fault_collection.py [hunk 1]
```diff
@@ -21,67 +21,66 @@
 comment_regexp = re.compile("--.*")
 
 def dequote(s):
-    if s[0] in ["'" , '"']:
+    if s[0] in ["'", '"']:
         if s[0] == s[-1]:
             return s[1:-1]
         else:
-            raise ValueError("Quote fuckup")
+            raise ValueError("Failed to de-quote while loading faults.")
```
Comment (joakim-hove) [inline python/python/ert/ecl/faults/fault_collection.py]: I want my profanities :smile:

File: python/tests/core/ecl/test_pylint.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+#  for more details.
+
+import fnmatch
+import os
+import unittest
+from pylint import epylint as lint
+from ert.test import ExtendedTestCase
+
+class CodeTestCase(ExtendedTestCase):
```
Comment (joakim-hove) [inline python/tests/core/ecl/test_pylint.py]: I guess this file could be split a test class which could go in `python/python/ert/test/code_test.py` and the actual invocation which could go in a normal test class, without explicit `__name__ == '__main__'` code?

Comment (pgdr) [inline python/tests/core/ecl/test_pylint.py]: I have removed `__name__...` — it wasn't in use.

**Comment by pgdr** [comment]
All the tests pass except the `ping` test, and I'm not completely sure what I've done to change that (except rewriting `.travis.yml`)

**Comment by pgdr** [comment]
If you look at [the last built master](https://travis-ci.org/Statoil/libecl/jobs/223141766), you can see that there are 142 tests run, compared to this branch, which has 144.  The two differences are util_ping and pylint.

**Comment by joakim-hove** [comment]
> All the tests pass except the ping test, and I'm not completely sure what I've done to change that

The ping testing + configuration has been long broken; ignore it.

File: python/tests/core/ecl/test_pylint.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        for folder in paths:
+            for root, _, filenames in os.walk(folder):
+                for filename in fnmatch.filter(filenames, '*.py'):
+                    if filename not in whitelist:
+                        matches.append(os.path.join(root, filename))
+        return matches
+
+    def test_linting(self):
+        paths = ['ert/ecl', 'ert/geo', 'ert/util', 'ert/well']  # TODO should be ert, cwrap, tests
+        white = ['ecl_kw.py', 'ecl_type.py', 'ecl_sum.py', 'ecl_grid.py', 'ecl_npv.py']  # TODO fix issues and remove
```
Comment (joakim-hove) [inline python/tests/core/ecl/test_pylint.py]: To me this seems more like a *black* list than a white list?

Comment (pgdr) [inline python/tests/core/ecl/test_pylint.py]: It's a _whitelist_ in pylint's terms: the package in question is allowed to have errors.

**Comment by joakim-hove** [comment]
Could the ping issue be which infrastructure travis is using?

**Comment by joakim-hove** [comment]
I suggest we merge #6 - which I think should be merged anyway, and then we complete this PR.

**Comment by pgdr** [comment]
> I suggest we merge #6 - which I think should be merged anyway, and then we complete this PR.

Okay, I've approved it.

**Comment by pgdr** [comment]
> I suggest we merge #6 - which I think should be merged anyway, and then we complete this PR.

Okay, I've merged it.

File: python/tests/core/ecl/test_pylint.py (hunk 3 above)
Comment (joakim-hove) [inline python/tests/core/ecl/test_pylint.py]: OK; maybe a comment then?

File: python/python/ert/ecl/faults/fault_collection.py [hunk 4]
```diff
@@ -31,57 +31,56 @@ def dequote(s):
 
 
 class FaultCollection(object):
-    def __init__(self , grid = None , *file_list):
+    def __init__(self, grid=None, *file_list):
```
Comment (joakim-hove) [inline python/python/ert/ecl/faults/fault_collection.py]: This code now seems to mix the input argument `grid` and the member variable `self.__grid` somewhat randomly. Is the member variable `self.__grid` at all needed?

File: python/tests/core/ecl/test_pylint.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+from pylint import epylint as lint
+from ert.test import ExtendedTestCase
+
+class CodeTestCase(ExtendedTestCase):
+    """This class is a test case for linting."""
+
+    LINT_ARGS = ['-d', 'R,C,W'] + \
+                ['--extension-pkg-whitelist=numpy']
+
+    def _get_lintable_files(self, paths, whitelist=None):
```
Comment (joakim-hove) [inline python/tests/core/ecl/test_pylint.py]: I would suggest one of two:

1. The two methods `_get_lintable_files()` and `test_linting()` are combined into one method:

```python
def lint_path(self, path, white_list):
     for root, _ . filenames = os.walk( path )
          ...
          # The lint call one single file could maybe be a separate method?
          self.assertEqual( 0 , lint.lint( )) 
```
And then an instance of `CodeTestCase` would call the `lint_path( )` method.

2. Or alternatively the `_get_lintable_files()` method should be `@classmethod` - currently it does not access `self`.

**Comment by joakim-hove** [comment]
OK - disregarding the ping trouble for now this is mostly good; and I would like to merge it. However I really want the `pylint` based test case to be in e.g. `ert/test/lint_testcase.py` - that way the `pylint` based testcase will be reusable for the other repositories in the ert ecosystem[1]; sorry to be nagging so much about this - the current approach just rubs me a bit the wrong way :-(

[1]: I guess it could inherit just from plain `unittest.TestCase`and be even more reusable?

**Comment by joakim-hove** [comment]
> ping trouble

Maybe try to remove the `test_assert_true( util_ping("localhost") );`?

File: python/python/ert/ecl/faults/fault_collection.py (hunk 4 above)
Comment (pgdr) [inline python/python/ert/ecl/faults/fault_collection.py]: fixed.

**Comment by joakim-hove** [comment]
Good work!

**Comment by pgdr** [comment]
Ok, tests pass, have approval, this one goes in.

**Pull request merged**